service/history/replication/executable_task.go

```diff
@@ -431,11 +438,71 @@ func (e *ExecutableTaskImpl) Resend(
 
 func (e *ExecutableTaskImpl) SyncState(
 	ctx context.Context,
-	remoteCluster string,
 	syncStateErr *serviceerrors.SyncState,
 	remainingAttempt int,
 ) (bool, error) {
-	return false, serviceerror.NewUnimplemented("not implemented") // TODO
+
+	remoteAdminClient, err := e.ClientBean.GetRemoteAdminClient(e.sourceClusterName)
+	if err != nil {
+		return false, err
+	}
+
+	targetClusterInfo := e.ClusterMetadata.GetAllClusterInfo()[e.ClusterMetadata.GetCurrentClusterName()]
+	_, err = remoteAdminClient.SyncWorkflowState(ctx, &adminservice.SyncWorkflowStateRequest{
+		NamespaceId: syncStateErr.NamespaceId,
+		Execution: &commonpb.WorkflowExecution{
+			WorkflowId: syncStateErr.WorkflowId,
+			RunId:      syncStateErr.RunId,
+		},
+		VersionedTransition: syncStateErr.VersionedTransition,
+		// TODO: add version histories to serviceerrors.SyncState and use it here.
+		// VersionHistories: nil,
+		TargetClusterId: int32(targetClusterInfo.InitialFailoverVersion),
```

Comment:
no.. I tracked the usage of that field and believe InitialFailoverVersion is the right one to use. Also `targetClusterInfo.ClusterID` is a string, while the `TargetClusterId` in the request is a int32. 

This is actually very confusing. I think it's because the streaming logic is using initialFailoverVersion and named it ClusterID.

Reply:
Thanks for the info. The naming is confusing as one name refers to different things.